src/NServiceBus.NHibernate.AcceptanceTests/When_using_synchronized_session_via_container.cs

```diff
@@ -0,0 +1,89 @@
+﻿using System.Threading.Tasks;
+using NServiceBus.AcceptanceTesting;
+using NServiceBus.AcceptanceTests.EndpointTemplates;
+using NUnit.Framework;
+
+namespace NServiceBus.AcceptanceTests.Sagas
+{
+    using global::NHibernate;
+
+    [TestFixture]
+    public class When_using_synchronized_session_via_container : NServiceBusAcceptanceTest
+    {
+        [Test]
+        public async Task Should_inject_synchronized_session_into_handler()
+        {
+            var context = await Scenario.Define<Context>()
+                .WithEndpoint<Endpoint>(b => b.When(s => s.SendLocal(new MyMessage())))
+                .Done(c => c.Done)
+                .Run()
+                .ConfigureAwait(false);
+
+            Assert.True(context.RepositoryHasSession);
+        }
+
+        public class Context : ScenarioContext
+        {
+            public bool Done { get; set; }
+            public bool RepositoryHasSession { get; set; }
+        }
+
+        public class Endpoint : EndpointConfigurationBuilder
+        {
+            public Endpoint()
+            {
+                EndpointSetup<DefaultServer>(c =>
+                {
+                    c.RegisterComponents(cc =>
+                    {
+                        cc.ConfigureComponent<MyRepository>(DependencyLifecycle.InstancePerUnitOfWork);
+                        cc.ConfigureComponent(b => b.Build<INHibernateStorageSession>().Session, DependencyLifecycle.InstancePerUnitOfWork);
+                    });
+                });
+            }
+
+            public class MyMessageHandler : IHandleMessages<MyMessage>
```

Comment:
Should we add another handler and test that session is the same for the whole `UnitOfWork`?

Reply:
Would it be enough to call `GetHashCode` and ensure both sessions are actually the same instance?

Reply:
I was thinking about this case a there were some workarounds needed (asynclocal) to make this work. And yes I think that `GetHashCode` is enough.